lib/fine-types-test/Setup.hs

```diff
@@ -0,0 +1,30 @@
+import Control.Monad
+import Distribution.Simple
+import Distribution.Simple.BuildPaths (autogenComponentModulesDir)
+import Distribution.Simple.LocalBuildInfo
+import System.Directory
+import System.FilePath
+import System.Process (callProcess)
+
+main :: IO ()
+main = defaultMainWithHooks hooks
+
+hooks :: UserHooks
+hooks = simpleUserHooks{postConf = \_ _ _ lbi -> generateSources lbi}
+
+generateSources :: LocalBuildInfo -> IO ()
+generateSources lbi =
+    forM_ (allComponentsInBuildOrder lbi) $ \clbi -> do
+        let dir = autogenComponentModulesDir lbi clbi
+            file = dir </> "Language/FineTypes/Test/UTxO.hs"
```

Comment:
Todo: Make this more general, work with any `LocalBuildInfo`, not just this particular file.

Reply:
To be done in a separate pull request. ADP-3180